lib/task/import/validate/csvValidatorCollection.php

```diff
@@ -0,0 +1,141 @@
+    {
+        foreach ($this->validators as $validator) {
+            $validator->setFilename($filename);
+        }
+    }
+
+    public function setColumnCount(int $longestRow)
+    {
+        foreach ($this->validators as $validator) {
+            $validator->setColumnCount($longestRow);
+        }
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        foreach ($this->validators as $validator) {
+            $validator->testRow($header, $row);
+        }
+    }
+
+    public function getResultCollection(): CsvValidatorResultCollection
+    {
+        $resultCollection = new CsvValidatorResultCollection();
+
+        foreach ($this->validators as $validator) {
+            $resultCollection->appendResult($validator->getTestResult());
+        }
+
+        return $resultCollection;
+    }
+
+    /**
+     * Gets an instantiated validatorCollection object populated with
+     * the correct Validator objects for the specified $className.
+     *
+     * @param string $className             The entity name. e.g. QubitInformationObject
+     * @param array  $options               the options array passed to the validator
+     * @param string $specificValidatorList comma separated string of validator class names
+     */
+    public static function getValidatorCollection(string $className, array &$options, ?string $validatorList = null): CsvValidatorCollection
+    {
+        // Map of available validator classes.
+        $validatorsAvailable = [
+            'CsvSampleValuesValidator' => CsvSampleValuesValidator::class,
+            'CsvFileEncodingValidator' => CsvFileEncodingValidator::class,
+            'CsvColumnNameValidator' => CsvColumnNameValidator::class,
+            'CsvColumnCountValidator' => CsvColumnCountValidator::class,
+            'CsvDuplicateColumnNameValidator' => CsvDuplicateColumnNameValidator::class,
+            'CsvEmptyRowValidator' => CsvEmptyRowValidator::class,
+            'CsvCultureValidator' => CsvCultureValidator::class,
+            'CsvLanguageValidator' => CsvLanguageValidator::class,
+            'CsvFieldLengthValidator' => CsvFieldLengthValidator::class,
+            'CsvParentValidator' => CsvParentValidator::class,
+            'CsvLegacyIdValidator' => CsvLegacyIdValidator::class,
+            'CsvEventValuesValidator' => CsvEventValuesValidator::class,
+            'CsvDigitalObjectPathValidator' => CsvDigitalObjectPathValidator::class,
+            'CsvDigitalObjectUriValidator' => CsvDigitalObjectUriValidator::class,
+        ];
+
+        // Remove tests that are not applicable to this $className.
+        foreach ($validatorsAvailable as $key => $class) {
+            if (!in_array($className, $class::LIMIT_TO) && !empty($class::LIMIT_TO)) {
+                unset($validatorsAvailable[$key]);
+            }
+        }
+
+        // If specific tests not used, use default validators for this $className.
+        if (!isset($validatorList)) {
```

Comment:
I think `if (empty($validatorList))` might be better here, because it evaluates true when `$validatorList` is an empty string.

Reply:
Yes -  `empty()` is better and handles this edge case more cleanly. Changed in next commit.